src/service/token-manager/token-manager.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Token } from './token-manager.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenManagerService {
+  private TOKEN = 'TOKEN';
+
+  storeToken(token: Token): Token {
+    localStorage.setItem(this.TOKEN, JSON.stringify(token));
+    return token;
+  }
+
+  getToken(): Token {
```

Comment:
Lepszą sygnaturą nie byłoby `getToken(): Token | null` ? Luźna propozycja. Tak samo w przypadku błędu nie zwracałbym `emptyToken` tylko rzucił błędem swoim / dodał tylko komentarz, że potem trzeba to inaczej obsłużyć (np. wysłać message do monitorach błędów jak Sentry). Taka próba "ładnego" obsłużenia błędu może narobić więcej szkody niż pożytku. Jeśli `JSON.parse` rzuca tutaj błędem to coś ostro musiało pójść nie tak i mamy buga w aplikacji. To oznacza, że aplikacja powinna przerwać co robi (np .requesty do których dodawane są te tokeny nie powinny się odbyć z pustym tokenem), zarejestrować buga i powinno się go naprawić :)